addons/sale/models/sale.py

```diff
@@ -511,13 +511,19 @@ def create(self, vals):
         return result
 
     def _compute_field_value(self, field):
+        send_mail = True
+        for record in self:
+            if field.name == 'invoice_status' and record._origin.invoice_status == 'upselling':
+                send_mail = False
+
         super()._compute_field_value(field)
-        if field.name != 'invoice_status' or self.env.context.get('mail_activity_automation_skip'):
+        if field.name != 'invoice_status' or self.env.context.get('mail_activity_automation_skip') or not send_mail:
             return
```

Comment:
So if I understand you logic, as soon as 1 of the record satisfies your condition we ignore the whole process for the whole recordset?

I think that the solution is somewhere else 🙂
I am sure you will find it!

Reply:
I've set up the linter as suggested in the Odoo documentation but i have this kind of error in file that i never edited
`continuation line over-indented for hanging indentpycodestyle(E126)` should i also ignore them ?

Reply:
Yep, only apply linter changes into your code changes, it does not worth applying it were you do not edit the code.

Reply:
I think the _origin is updated after the super().compute_field_value() so the only place where i can check if the invoice status is being modified

Reply:
yep you are right, then you should try to compute your subset of recordset prior to it, but surely not skip the whole process if one meets the condition 🙂